Apps/WebClient/src/Server/Controllers/UserProfileController.cs

```diff
@@ -184,6 +184,7 @@ public IActionResult RecoverUserProfile(string hdid)
         [HttpGet]
         [Route("termsofservice")]
         [AllowAnonymous]
+        [ResponseCache(Location = ResponseCacheLocation.Any, Duration = 3600)]
```

Comment:
- On the top of caching at the client and proxy level, I think the best method of caching is to use the in-memory cache. However, as we don't expect that the TermsOfService page will be requested very often, so it doesn't make sense to loose memory to cache it in this scenario.
- ref. https://docs.microsoft.com/en-us/aspnet/core/performance/caching/response

Reply:
pretty cool. Have you checked if the response header says is cached?

Reply:
I run the WebClient in VS 2019 debug mode then open the ToS page in Chrome then click F5. The first load should call the controller method. The 2nd one should not.

Reply:
yeah, the header response on chrome dev tools would also say is cache, but that is a good test too!

Reply:
ah great to know this header response, thanks @FuriousLlama